openhtf/output/web_gui/__init__.py

```diff
@@ -223,7 +222,7 @@ def __getitem__(self, hostport):  # pylint:disable=invalid-name
     """Provide dictionary-like access to the station store."""
     if not isinstance(hostport, Hostport):
       raise ValueError('StationStore key must be a Hostport instance.')
-    return self.stations.setdefault(hostport, None)
+    return self.stations.get(hostport)
```

Comment:
Right—I couldn't figure out why we'd ever want to store `None` in `self.stations`.

It has the *potential* to cause an insidious bug:

Currently the default doesn't matter because it's never actually triggered. But suppose it were, because the way StationStore is implemented suggests that it's okay to access it with missing keys. For example, in `StationPubSub.on_subscribe` change
```python
if hostport not in self._store.stations:
```
to
```python
if not self._store.stations[hostport]:
```

Then the server will crash because `DashboardPubSub.publish_discovery_update` assumes the values are never `None`. That's a good thing. Suppose instead we thought `None` values are okay. For example, in `DashboardPubSub.publish_discovery_update` add
```python
if station is None:
  continue
```

Then it will fail silently on certain edge cases because you can't discover a station if its hostport is already assigned to `None`.